.NET/Microsoft.Recognizers.Text.DateTime/Parsers/BaseDateParser.cs

```diff
@@ -292,51 +292,27 @@ private DateTimeResolutionResult ParseImplicitDate(string text, DateObject refer
                 ret.Success = true;
 
                 return ret;
-            }
-
+            }
+
             // handling cases like 'Thursday the 21st', which both 'Thursday' and '21st' refer to a same date
-            match = this.config.WeekDayAndDayOfMothRegex.Match(text);
-            if (match.Success)
+            match = this.config.WeekDayAndDayOfMothRegex.Match(text);
+            if (match.Success)
             {
                 int month = referenceDate.Month, year = referenceDate.Year;
-
                 // create a extract result which content ordinal string of text
                 ExtractResult erTmp = new ExtractResult();
                 erTmp.Text = match.Groups["DayOfMonth"].Value.ToString();
                 erTmp.Start = match.Groups["DayOfMonth"].Index;
                 erTmp.Length = match.Groups["DayOfMonth"].Length;
-
-                // parse the day in text into number, and get week of day of this number which regarded as a day in this month
+                // parse the day in text into number
                 var day = Convert.ToInt32((double)(this.config.NumberParser.Parse(erTmp).Value ?? 0));
-                var date = DateObject.MinValue.SafeCreateFromValue(referenceDate.Year, referenceDate.Month, day);
-                var date2weekdayStr = date.DayOfWeek.ToString().ToLower();
-
-                // get week day from text directly, compare it with the weekday parse above
-                // to see whether they refer to a same week day
-                var extractedWeekDayStr = match.Groups["weekday"].Value.ToString().ToLower();
-                if (!date.Equals(DateObject.MinValue) &&
-                    config.DayOfWeek[date2weekdayStr] == config.DayOfWeek[extractedWeekDayStr])
-                {
-                    ret.Timex = FormatUtil.LuisDate(year, month, day);
-                    ret.FutureValue = new DateObject(year, month, day); ;
-                    ret.PastValue = new DateObject(year, month, day); ;
-                    ret.Success = true;
-
-                    return ret;
-                }
-                else
-                {
-                    //the resolution should have no value
-                    var weekdayStr = match.Groups["weekday"].Value.ToLower();
-                    var weekDay = this.config.DayOfWeek[weekdayStr];
-                    var timexDayOfWeek="XXXX-WXX-" + weekDay;

```

Comment:
Where did this go now?

Reply:
we determine that in the extractor. We will output two separate results instead of a whole with no resolution when the two dates are not match.